File: festival-web/src/main/java/cn/bdqfork/web/route/annotation/RouteMapping.java
package cn.bdqfork.web.route.annotation;

import io.vertx.core.http.HttpMethod;

import java.lang.annotation.*;

/**
 * 用于将url请求映射到方法
 *
 * @author bdq
 * @since 2020/1/21
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE, ElementType.METHOD})
public @interface RouteMapping {
    String value() default "";

    HttpMethod method() default HttpMethod.OPTIONS;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/annotation/RouteMapping.kt
package cn.bdqfork.kotlin.web.route.annotation

import io.vertx.core.http.HttpMethod
import io.vertx.ext.web.handler.TimeoutHandler

/**
 * 用于将url请求映射到方法
 *
 * @author bdq
 * @since 2020/1/21
 */
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
annotation class RouteMapping(val value: String = "", val method: HttpMethod = HttpMethod.OPTIONS, val timeout: Long = TimeoutHandler.DEFAULT_TIMEOUT)